client/apollo/react/src/Message/MessageCommon.tsx

```diff
@@ -1,68 +1,57 @@
-import validationIcon from "@material-symbols/svg-400/outlined/check_circle-fill.svg";
-import errorIcon from "@material-symbols/svg-400/outlined/error-fill.svg";
-import warningIcon from "@material-symbols/svg-400/outlined/warning-fill.svg";
-import neutralIcon from "@material-symbols/svg-400/outlined/info-fill.svg";
-import infoIcon from "@material-symbols/svg-400/outlined/emoji_objects-fill.svg";
 import {
   type ComponentPropsWithoutRef,
   ComponentType,
-  type PropsWithChildren,
   type ReactElement,
-  useMemo,
+  type ReactNode,
 } from "react";
-
 import type { ButtonProps } from "../Button/ButtonCommon";
+import type { Icon } from "../Icon/IconCommon";
 import { Link } from "../Link/LinkCommon";
-import { Svg } from "../Svg/Svg";
+import { getComponentClassName } from "../utilities/getComponentClassName";
+import { messageVariants } from "./constants";
+import { getAriaRole, getIconFromVariant } from "./Message.helpers";
+import type { MessageVariants } from "./types";
 
 type Headings = "h2" | "h3" | "h4" | "h5" | "h6";
 
-export const messageVariants = {
-  validation: "validation",
-  error: "error",
-  warning: "warning",
-  information: "information",
-  neutral: "neutral",
-} as const;
-
-export type MessageVariants = keyof typeof messageVariants;
-
 export type MessageProps = {
+  /** Message variant (validation, error, warning, information, neutral) */
   variant: MessageVariants;
+  /** Title displayed in the message */
   title?: string;
+  /** Main content of the message */
+  children?: ReactNode;
+  /** Action (link or button) displayed on the right side of the message */
   action?: ReactElement<typeof Link | ComponentType<ButtonProps>>;
+  /** Icon size in pixels */
   iconSize?: number;
+  /** HTML heading level used for the title */
   heading?: Headings;
 } & ComponentPropsWithoutRef<"div">;
 
-const getIconFromType = (variant: MessageVariants) =>
-  ({
-    [messageVariants.validation]: validationIcon,
-    [messageVariants.error]: errorIcon,
-    [messageVariants.neutral]: neutralIcon,
-    [messageVariants.warning]: warningIcon,
-    [messageVariants.information]: infoIcon,
-  })[variant] || infoIcon;
+type MessagePropsWithComponents = MessageProps & {
+  IconComponent: typeof Icon;
+};
 
-export const Message = ({
+export const MessageCommon = ({
   variant = messageVariants.information,
   className,
   title,
   children,
   action,
   iconSize = 24,
   heading: Heading = "h4",
-}: PropsWithChildren<MessageProps>) => {
-  const icon = useMemo(() => getIconFromType(variant), [variant]);
+  IconComponent,
+}: MessagePropsWithComponents) => {
+  const icon = getIconFromVariant(variant);
+  const role = getAriaRole(variant);
 
   return (
-    <div
-      className={["af-message", `af-message--${variant}`, className]
-        .filter(Boolean)
-        .join(" ")}
-      role="alert"
+    <section
+      className={getComponentClassName("af-message", className, variant)}
```

Comment:
je ne suis pas fan ce de cette méthode je préfère l'ancienne syntaxe, de plus cette méthone ne place pas le className de l'utilisateur en fin de string ce qui n'est pas bon pour moi.

au pire on peut faire 
```suggestion
      className={classNames("af-message", `af-message--${variant}`, className)}
```
qui est bien plus propre selon moi

Reply:
Je remplacé par getComponentClassName car selon moi il faudrait tout le temps utiliser une seule et même fonction de formatage de classes afin d'uniformiser les usages.
Si tu veux je peux faire une autre PR avec cette modification : 
<img width="598" height="605" alt="image" src="https://github.com/user-attachments/assets/7c2399e1-2ad3-4fc2-bf90-3312f6ce57eb" />

Car je suis plutôt d'accord sur le fait de mettre les classes ajoutées par le dev à la suite car c'est un ajout.

Reply:
je suis d'accord pour d'uniformiser mais getComponentClassName pour moi me semble overkill, et surtout ca me rappel de vieux souvenir de slash grrr.

Reply:
Ce que je peux faire c'est de le gérer manuellement et on prendra une décision en point de synchro.
(le soucis c'est que je ne pense pas pouvoir participer à un point de synchro avant un moment)

Reply:
Fixed in https://github.com/AxaFrance/design-system/pull/1373/commits/a266e7f1dff8d8b6673f26593cd6df6c371fd1ee